File: BackEnd/src/main/java/com/company/entity/repair/Reservation.java
package com.company.entity.repair;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import java.time.LocalDateTime;

@Getter
@Setter
@Entity
@Table(name = "reservations")
public class Reservation {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "repair_store_id", nullable = false)
    private Long repairStoreId;

    @Column(name = "user_id", nullable = false)
    private String userId;

    @Column(name = "car_id", nullable = false)
    private Long carId;

    @Column(name = "reservation_time", nullable = false)
    private LocalDateTime reservationTime;

    @Column(name = "details")
    private String details;

    @Enumerated(EnumType.STRING)
    @Column(name = "status", nullable = false)
    private ReservationStatus status;

    @Column(name = "created_at", nullable = false, updatable = false,
            insertable = false, columnDefinition = "TIMESTAMP DEFAULT CURRENT_TIMESTAMP")
    private LocalDateTime createdAt;

    @PrePersist
    public void onPrePersist() {
        if (status == null) {
            status = ReservationStatus.PENDING;
        }
    }
}

File: GaragEZ/src/types/reservation.ts
// src/types/Reservation.ts
export interface Reservation {
  id: number;
  repairStoreId: number;
  userId: string;
  carId: number;
  reservationTime: string; // "2025-03-20T10:00:00" 등
  details: string;
  status: string;          // "PENDING" | "CONFIRMED" | "CANCELLED"
  createdAt: string;       // "2025-03-01T09:30:00"
}
